opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -11,3 +11,365 @@
+                           ) -> 'MeasureInt':
+        """Creates a measure used to record raw int values.
+
+        Args:
+            name: The name of the measure.
+            description: Human readable description of this measure.
+            unit: Unit of the measure values.
+            label_keys: list of keys for the labels with dynamic values.
+                Order of the list is important as the same order MUST be used
+                on recording when suppling values for these labels.
+            aggregation: The type of aggregation to use for this measure metric.
+            span_context: The `SpanContext` that identified the `Span`
+                for which the metrics are associated with.
+
+        Returns:
+            A new `MeasureInt`
+        """
+
+    def create_float_measure(self,
+                             name: str,
+                             description: str,
+                             unit: str,
+                             label_keys: LabelKeys,
+                             aggregation: 'Aggregation',
+                             span_context: SpanContext = None,
+                             ) -> 'MeasureFloat':
+        """Creates a Measure used to record raw float values.
+
+        Args:
+            name: the name of the measure
+            description: Human readable description of this measure.
+            unit: Unit of the measure values.
+            label_keys: list of keys for the labels with dynamic values.
+                Order of the list is important as the same order MUST be used
+                on recording when suppling values for these labels.
+            aggregation: The type of aggregation to use for this measure metric.
+            span_context: The `SpanContext` that identified the `Span`
+                for which the metrics are associated with.
+
+        Returns:
+            A new `MeasureFloat`
+        """
+
+
+class Metric(ABC):
+    """Base class for various types of metrics.
+
+    Metric class that inherit from this class are specialized with the type of
+    time series that the metric holds. Metric is constructed from the meter.
+    """
+
+    @abstractmethod
+    def get_or_create_time_series(self,
+                                  label_values: LabelValues
+                                  ) -> 'object':
+        """Gets and returns a timeseries, a container for a cumulative value.
+
+        If the provided label values are not already associated with this
+        metric, a new timeseries is returned, otherwise it returns the existing
+        timeseries with the exact label values. The timeseries returned
+        contains logic and behaviour specific to the type of metric that
+        overrides this function.
+
+        Args:
+            label_values: A map of `LabelValue` s that will be associated
+                with the return timeseries.
+        """
+
+    @abstractmethod
+    def get_default_time_series(self) -> 'object':
```

Comment:
It seems like this could return a `TimeSeries` (or whatever class you end up using). If you're putting the `Metric` class hierarchy in the API, why not the container types they use too?